library/src/main/java/com/google/maps/android/clustering/ClusterManager.java

```diff
@@ -156,44 +160,95 @@ public void clearItems() {
         }
     }
 
-    public void addItems(Collection<T> items) {
+    /**
+     * Adds items to clusters. After calling this method you must invoke {@link #cluster()} for the
+     * state of the clusters to be updated on the map.
+     * @param items items to add to clusters
+     * @return true if the cluster manager contents changed as a result of the call
+     */
+    public boolean addItems(Collection<T> items) {
+        boolean result;
         mAlgorithm.lock();
         try {
-            mAlgorithm.addItems(items);
+            result = mAlgorithm.addItems(items);
```

Comment:
The `result` variable is not needed—the block within `finally` will still be invoked even after a `return`.

Reply:
Oh, good point - thanks for pointing this out. I just removed these for all the relevant methods and returned the result of `mAlgorithm.x()` directly in https://github.com/googlemaps/android-maps-utils/pull/627/commits/45af836aba6a1871a60150ee346c2bb183b298e6. Much cleaner.